test/types/peg.test-d.ts

```diff
@@ -0,0 +1,330 @@
+        add(node.type);
+        expectType<peggy.ast.Suffixed>(node);
+        expectType<"optional" | "zero_or_more" | "one_or_more">(node.type);
+        expect(node.type).toBe("zero_or_more");
+        expectType<peggy.LocationRange>(node.location);
+        expectType<peggy.ast.Primary>(node.expression);
+        visit(node.expression);
+      },
+      one_or_more(node) {
+        add(node.type);
+        expectType<peggy.ast.Suffixed>(node);
+        expectType<"optional" | "zero_or_more" | "one_or_more">(node.type);
+        expect(node.type).toBe("one_or_more");
+        expectType<peggy.LocationRange>(node.location);
+        expectType<peggy.ast.Primary>(node.expression);
+        visit(node.expression);
+      },
+      group(node) {
+        add(node.type);
+        expectType<peggy.ast.Group>(node);
+        expectType<"group">(node.type);
+        expectType<peggy.LocationRange>(node.location);
+        expectType<peggy.ast.Labeled | peggy.ast.Sequence>(node.expression);
+        visit(node.expression);
+      },
+      semantic_and(node) {
+        add(node.type);
+        expectType<peggy.ast.SemanticPredicate>(node);
+        expectType<"semantic_and" | "semantic_not">(node.type);
+        expect(node.type).toBe("semantic_and");
+        expectType<peggy.LocationRange>(node.location);
+        expectType<string>(node.code);
+        expectType<peggy.LocationRange>(node.codeLocation);
+      },
+      semantic_not(node) {
+        add(node.type);
+        expectType<peggy.ast.SemanticPredicate>(node);
+        expectType<"semantic_and" | "semantic_not">(node.type);
+        expect(node.type).toBe("semantic_not");
+        expectType<peggy.LocationRange>(node.location);
+        expectType<string>(node.code);
+        expectType<peggy.LocationRange>(node.codeLocation);
+      },
+      rule_ref(node) {
+        add(node.type);
+        expectType<peggy.ast.RuleReference>(node);
+        expectType<"rule_ref">(node.type);
+        expectType<peggy.LocationRange>(node.location);
+        expectType<string>(node.name);
+      },
+      literal(node) {
+        add(node.type);
+        expectType<peggy.ast.Literal>(node);
+        expectType<"literal">(node.type);
+        expectType<peggy.LocationRange>(node.location);
+        expectType<string>(node.value);
+        expectType<boolean>(node.ignoreCase);
+      },
+      class(node) {
+        add(node.type);
+        expectType<peggy.ast.CharacterClass>(node);
+        expectType<"class">(node.type);
+        expectType<peggy.LocationRange>(node.location);
+        expectType<boolean>(node.inverted);
+        expectType<boolean>(node.ignoreCase)
+        expectType<(string | string[])[]>(node.parts);
+      },
+      any(node) {
+        add(node.type);
+        expectType<peggy.ast.Any>(node);
+        expectType<"any">(node.type);
+        expectType<peggy.LocationRange>(node.location);
+      }
+    });
+
+    // @dts-jest:pass:snap
+    visit(grammar);
```

Comment:
If I understand correctly, that construction mean to run `visit(grammar)` and assert types with snapshot that was taken on the first run. However that snapshot should be committed and it seems it is not committed

Reply:
This is a mistake, left over from a previous attempt to use dts-jest.  Will remove.